File: libs/std/basicMath.js
import {SltFunc, SltThunk, SltNum} from '../../SltRuntime.ts'

const unsafeExponent = new SltThunk(
    () =>
        new SltFunc(
            a =>
                new SltFunc(
                    b => new SltNum(a().value ** b().value, a().pos),
                    [7, 1, "intify.lua"]
                ),
                [7, 1, "intify.lua"]
            ),
)

const unsafeCeil = new SltThunk(
    () => 
        new SltFunc(
            a => new SltNum(Math.ceil(a().value), a().pos),
            [7, 1, "intify.lua"]
        )
)

const unsafeFloor = new SltThunk(
    () => 
        new SltFunc(
            a => new SltNum(Math.floor(a().value), a().pos),
            [7, 1, "intify.lua"]
        )
)

const unsafeRound = new SltThunk(
    () => 
        new SltFunc(
            a => new SltNum(Math.round(a().value), a().pos),
            [7, 1, "intify.lua"]
        )
)

const unsafeMod = 
  new SltThunk(
    () => {
      return new SltFunc(
        a =>
          new SltFunc(
            b => new SltNum(a().value % b().value, b().loc),
            a().loc
          ),
          [0, 0, "core"]
        )
      }
    )

export {unsafeExponent, unsafeRound, unsafeCeil, unsafeFloor, unsafeMod}

File: libs/std/basicMath.lua
require 'SltRuntime'

function round(x)
    return x>=0 and math.floor(x+0.5) or math.ceil(x-0.5)
end

unsafeRound = SltThunk.create(
    function() return
        SltFunc.create(
            function(a) return
                    SltNum.create(round(a().value), a().pos)
            end,
            {7, 1, "intify.lua"}
        )
    end
)

unsafeCeil = SltThunk.create(
    function() return 
        SltFunc.create(
            function(a) return SltNum.create(math.ceil(a().value), a().pos) end
        )
    end
)

unsafeFloor = SltThunk.create(
    function() return 
        SltFunc.create(
            function(a) return SltNum.create(math.floor(a().value), a().pos) end
        )
    end
)

unsafeExponent = SltThunk.create(
    function() return
        SltFunc.create(
            function(a) return
                SltFunc.create(
                    function(b) return
                        SltNum.create(math.pow(a().value, b().value), a().pos)
                    end
                )
            end
            ),
            {7, 1, "intify.lua"}
    end
)

unsafeMod = 
  SltThunk.create(
    function() return
      SltFunc.create(
        function(a)
          return SltFunc.create(
            function(b)
              return SltNum.create(math.fmod(a().value, b().value), b().loc)
            end
          )
        end
      )
    end
  )